platform/ui/src/hooks/useStorage.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export const useSessionStorage = (key, defaultValue = {}) => {
```

Comment:
These are neat. Can we JSDOC usage? Any easy way to unit test these?

Reply:
https://blog.logrocket.com/a-quick-guide-to-testing-react-hooks-fa584c415407/

We may also want a file per hook. I realize these are similar, but I would expect a `useStorage` hook in this file at first glance.